hydrus/data/resource_based_classes.py

```diff
@@ -54,99 +56,143 @@ def insert_object(object_: Dict[str, Any], session: scoped_session) -> str:
+            # put the id of the foreign instance in this table's column
+            object_[fk_column] = fk_object_id
         try:
-            fk_object = object_[fk_column]
-        except KeyError as e:
-            wrong_property = e.args[0]
-            raise PropertyNotGiven(type_=wrong_property)
-        # insert the foreign key object
-        fk_object_id = insert_object(fk_object, session)
-        # put the id of the foreign instance in this table's column
-        object_[fk_column] = fk_object_id
-    try:
-        # remove the @type from object before using the object to make a
-        # instance of it using sqlalchemy class
-        object_.pop("@type")
-        inserted_object = database_class(**object_)
-    except TypeError as e:
-        # extract the wrong property name from TypeError object
-        wrong_property = e.args[0].split("'")[1]
-        raise PropertyNotFound(type_=wrong_property)
-    try:
-        session.add(inserted_object)
-        session.commit()
-    except InvalidRequestError:
-        session.rollback()
+            # remove the @type from object before using the object to make a
+            # instance of it using sqlalchemy class
+            object_.pop("@type")
+            inserted_object = database_class(**object_)
+        except TypeError as e:
+            # extract the wrong property name from TypeError object
+            wrong_property = e.args[0].split("'")[1]
+            raise PropertyNotFound(type_=wrong_property)
+        try:
+            session.add(inserted_object)
+            session.commit()
+        except InvalidRequestError:
+            session.rollback()
 
-    return inserted_object.id
+        return inserted_object.id
 
 
 def get_object(
-    query_info: Dict[str, str], session: scoped_session
+    query_info: Dict[str, str], session: scoped_session, collection: bool = False
 ) -> Dict[str, str]:
     """
     Get the object from the database
     :param query_info: Dict containing the id and @type of object that has to retrieved
     :param session: sqlalchemy session
+    :param collection: True if the type_ is of a collection, False for any other class
     :return: dict of object with its properties
     """
     type_ = query_info["@type"]
     id_ = query_info["id_"]
     database_class = get_database_class(type_)
-    try:
-        object_ = (
-            session.query(database_class)
-            .filter(database_class.id == id_)
-            .one()
-        ).__dict__
-    except NoResultFound:
-        raise InstanceNotFound(type_=type_, id_=id_)
-    # Remove the unnecessary keys from the object retrieved from database
-    object_template = copy.deepcopy(object_)
-    object_template.pop("_sa_instance_state")
-    object_template.pop("id")
-    object_template["@type"] = query_info["@type"]
-    return object_template
-
-
-def delete_object(query_info: Dict[str, str], session: scoped_session) -> None:
+    if collection:
+        objects = (
```

Comment:
Better to move these things to separate functions and call them here instead of having long if else branches.

Reply:
Yes, will do